client/fingerprint/memory.go

```diff
@@ -24,19 +24,28 @@ func NewMemoryFingerprint(logger *log.Logger) Fingerprint {
 }
 
 func (f *MemoryFingerprint) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
-	memInfo, err := mem.VirtualMemory()
-	if err != nil {
-		f.logger.Printf("[WARN] Error reading memory information: %s", err)
-		return false, err
-	}
+	var totalMemory int
+
+	if cfg.MemoryMB != 0 {
+		totalMemory = cfg.MemoryMB
+	} else {
+		memInfo, err := mem.VirtualMemory()
+		if err != nil {
+			f.logger.Printf("[WARN] Error reading memory information: %s", err)
+			return false, err
+		}
 
-	if memInfo.Total > 0 {
-		node.Attributes["memory.totalbytes"] = fmt.Sprintf("%d", memInfo.Total)
+		if memInfo.Total > 0 {
+			node.Attributes["memory.totalbytes"] = fmt.Sprintf("%d", memInfo.Total)
```

Comment:
Also need to set memory.totalbytes as node attributes when its configured via MemoryMB

Reply:
done, see above